File: finding-seams/csharp/FindingSeams.Tests/ReminderSenderTest.cs
using FindingSeams.Reminders;
using Xunit;

namespace FindingSeams.Tests;

public class ReminderSenderTest
{
    /// <summary>Same wall-clock pressure as Mocha’s default 2000 ms test timeout.</summary>
    [Fact]
    public async Task Sends_reminders_qyuickly()
    {
        var reminderSender = new ReminderSender();

        await reminderSender.SendAllAsync(new List<Reminder> { new("1", "foo@example.com") })
            .WaitAsync(TimeSpan.FromSeconds(2));
    }
}

File: finding-seams/kotlin/src/test/kotlin/findingseams/ReminderSenderTest.kt
package findingseams

import findingseams.remindersender.Reminder
import findingseams.remindersender.ReminderSender
import kotlinx.coroutines.runBlocking
import kotlinx.coroutines.withTimeout
import org.junit.jupiter.api.Test

/** Same wall-clock pressure as Mocha’s default 2000 ms test timeout (see Python `asyncio.wait_for`). */
private const val REMINDER_TEST_TIMEOUT_MS = 2000L

class ReminderSenderTest {
    @Test
    fun `sends reminders qyuickly`() = runBlocking {
        val reminderSender = ReminderSender()

        withTimeout(REMINDER_TEST_TIMEOUT_MS) {
            reminderSender.sendAll(listOf(Reminder(id = "1", email = "foo@example.com")))
        }
    }
}
